addOns/automation/src/main/java/org/zaproxy/addon/automation/ExtensionAutomation.java

```diff
@@ -287,15 +290,21 @@ public void generateTemplateFile(String filename, boolean incAll) {
             }
 
             jobs.values().stream()
-                    .sorted()
+                    .filter(job -> job.getClass().getAnnotation(Deprecated.class) == null)
+                    .filter(Predicate.not(AutomationJob::isDataJob))
+                    .sorted(Comparator.comparing(AutomationJob::getType))
                     .forEach(
                             j -> {
                                 try {
-                                    if (incAll) {
-                                        fw.write(j.getTemplateDataMax());
-                                    } else {
-                                        fw.write(j.getTemplateDataMin());
+                                    String template =
+                                            incAll
+                                                    ? j.getTemplateDataMax()
+                                                    : j.getTemplateDataMin();
+                                    if (StringUtils.isBlank(template)) {
+                                        return;
                                     }
+                                    template = template.stripTrailing() + "\n\n";
```

Comment:
Doesn't this need to account for windows users manually editing the templates?

Reply:
They are bundled in the add-ons.